src/node/nodestate.h

```diff
@@ -519,15 +566,22 @@ namespace ccf
       }
       quote = quote_opt.value();
 #endif
-      join_rpc.params.quote = quote;
+      join_params.quote = quote;
 
       LOG_DEBUG_FMT(
         "Sending join request to {}:{}",
         args.config.joining.target_host,
         args.config.joining.target_port);
 
-      join_client->send_request(
-        join_rpc.method, jsonrpc::pack(join_rpc, jsonrpc::Pack::Text));
+      const auto body = jsonrpc::pack(join_params, jsonrpc::Pack::Text);
+
+      http::Request r(
```

Comment:
Should there be a generic constructor for `http::Request`s, i.e. `Request(const std::string& frontend, const std::string& method)`? It seems faffy to have to use `fmt::format()` every time.

Reply:
I don't think this is worth doing - the 2 instances in this file are the only places we manually do this. The clients construct it slightly differently. I'm also wary of formalising the concept of actor/frontend names elsewhere in the code.